kats/src/main/kotlin/kats/Reader.kt

```diff
@@ -0,0 +1,16 @@
+package kats
+
+/**
+ * Basic implementation of the Reader monad. Provides an "implicit" context (configuration) for
+ * function execution. Intended to provide Dependency Injection.
+ */
+class Reader<C, A>(val rd: (C) -> A) {
```

Comment:
I have added two pure methods inside of a `companion object Factory` right now to achieve two different ways of Lifting:
* One to lift a plain A value to a reader wrapping it inside of a function that takes no parameters at all.
* Another one to lift a read (`rd`) function to a Reader.